feat: add nop destination and cleanup deprecated components

blumamir opened the pull request:
This PR:

- removes the deprecated and unused logging exporter, which is replaced with the debug exporter which we do use.
- removed the deprecated `ballast` extension which  we don't use and is deprecated in favor of `GOMEMLIMIT` which we do use.
- adds the `nop` exporter which is useful for development and testing (for example - load scenarios)
- adds documentation in `CONTRIBUTING.md` regarding how to use the debug and nop destination for development.

edeNFed reviewed (approved):
Cool idea! 
Not: I think adding it as a target in Makefile will be more convenient than documenting in markdown file

blumamir commented:
> Cool idea! nit: I think adding it as a target in Makefile will be more convenient than documenting in markdown file

good idea. Added and updated the markdown
